sklearn/utils/testing.py

```diff
@@ -709,26 +709,13 @@ def set_random_state(estimator, random_state=0):
         estimator.set_params(random_state=random_state)
 
 
-def if_matplotlib(func):
-    """Test decorator that skips test if matplotlib not installed.
-
-    Parameters
-    ----------
-    func
-    """
-    @wraps(func)
-    def run_test(*args, **kwargs):
-        try:
-            import matplotlib
-            matplotlib.use('Agg', warn=False)
-            # this fails if no $DISPLAY specified
-            import matplotlib.pyplot as plt
-            plt.figure()
-        except ImportError:
-            raise SkipTest('Matplotlib not available.')
-        else:
-            return func(*args, **kwargs)
-    return run_test
+def has_matplotlib():
```

Comment:
we have a `check_matplotlib_support` util already

Reply:
`check_matplotlib_support` raise an error while I only need a boolean test.
Do you mean to call `check_matplotlib_support` within this helper?

Reply:
Ok yeah they're not doing the same thing.

Can we make it private to this file then? It's only really useful for the decorator